actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/util/UserUtil.java

```diff
@@ -245,18 +245,33 @@ public static void checkEmailUniqueness(String email) {
   }
 
   public static String getUserIdFromExternalId(Map<String, Object> userMap) {
+
     String extId = (String) userMap.get(JsonKey.EXTERNAL_ID);
     String provider = (String) userMap.get(JsonKey.EXTERNAL_ID_PROVIDER);
     String idType = (String) userMap.get(JsonKey.EXTERNAL_ID_TYPE);
     return userExternalIdentityService.getUser(extId, provider, idType);
   }
 
   public static String getUserId(Map<String, Object> userMap) {
-    String extId = (String) userMap.get(JsonKey.EXTERNAL_ID);
-    String provider = (String) userMap.get(JsonKey.EXTERNAL_ID_PROVIDER);
-    String idType = (String) userMap.get(JsonKey.EXTERNAL_ID_TYPE);
-    Map<String, String> providerOrgMap = fetchOrgIdByProvider(Arrays.asList(provider));
-    return userExternalIdentityService.getUser(extId, providerOrgMap.get(provider), idType);
+    String userId;
+    if (null != userMap.get(JsonKey.USER_ID)) {
+      userId = (String) userMap.get(JsonKey.USER_ID);
+    } else {
+      userId = (String) userMap.get(JsonKey.ID);
+    }
+    if (StringUtils.isBlank(userId)) {
```

Comment:
Missing ! eh?

Reply:
yes if we pass externalId information to get userId like earlier